torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -948,10 +948,10 @@ static void removeSequenceSplitConcat(Block* b) {
 // Work around limitation from ONNX that the block input cannot be used directly
 // as block output. Inserts an Identity node inside the block, and have the
 // block return the output of the Identity.
-static void insertIdentityForInputUsedAsOutput(Block* b) {
+void InsertIdentityForInputUsedAsOutput(Block* b) {
   for (auto out : b->outputs()) {
     auto n = out->node();
-    if (nullptr != n && n->kind() == prim::Param) {
+    if (out->node()->owningBlock() != b) {
```

Comment:
nit
```suggestion
    if (n->owningBlock() != b) {
```

Reply:
curious if the following are always true?
1. out->node() can never be nullptr?
2. prim::Param node for a certain block is not owned by that block?